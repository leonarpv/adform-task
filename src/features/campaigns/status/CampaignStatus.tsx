import React from 'react'
import classNames from 'classnames'
import './campaignStatus.scss'

type Props = {
    active: boolean
}

const CampaignStatus = ({ active }: Props) => {
    const badgeClassName = classNames(
        'campaign-status__badge',
        active
            ? 'campaign-status__badge--active'
            : 'campaign-status__badge--inactive'
    )

    return (
        <span className="campaign-status">
            <span className={badgeClassName} />

            {active ? 'Active' : 'Inactive'}
        </span>
    )
}

export default CampaignStatus
