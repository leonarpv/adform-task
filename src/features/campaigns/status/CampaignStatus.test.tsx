import { shallow } from 'enzyme'
import React from 'react'
import CampaignStatus from './CampaignStatus'

describe('CampaignStatus', () => {
    it('should render properly', () => {
        const active = true

        const wrapper = shallow(<CampaignStatus active={active} />)

        expect(wrapper).toMatchSnapshot()
    })

    it('should render status badge', () => {
        const active = true

        const wrapper = shallow(<CampaignStatus active={active} />)

        expect(
            wrapper
                .find('.campaign-status__badge.campaign-status__badge--active')
                .exists()
        ).toBe(true)
    })

    it('should render status badge in inactive state', () => {
        const active = false

        const wrapper = shallow(<CampaignStatus active={active} />)

        expect(
            wrapper
                .find(
                    '.campaign-status__badge.campaign-status__badge--inactive'
                )
                .exists()
        ).toBe(true)
    })
})
