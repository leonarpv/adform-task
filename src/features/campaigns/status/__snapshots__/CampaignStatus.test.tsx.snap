// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampaignStatus should render properly 1`] = `ShallowWrapper {}`;
